src/providers/docker-images/codebuild/linux-arm64/Dockerfile

```diff
@@ -65,3 +68,6 @@ VOLUME /var/lib/docker
 
 # configure runner
 USER runner
+
+ARG EXTRA_COMMANDS="echo 'no extra commands'"
+RUN bash -c "${EXTRA_COMMANDS}"
```

Comment:
Will this work with commands containing quotes? Can we maybe use an asset instead where the user just gives us a shell script to execute?

Reply:
with the quotes, it's tricky indeed, but I don't see any other way to use RUN with some arbitrary command like that
an asset is not ideal due to the intended (see example in docs) use case: ad-hoc operations like docker login

Reply:
We can have an inline asset for this purpose just like `aws_lambda.Code.fromInline`. I would prefer to not have gotchas like not supporting quotes in such a useful feature.

Reply:
how do I implement it then exactly? Do you have any examples?

Reply:
I can add more details later but basically:
```
    new s3_assets.Asset(this, 'Dependencies Definition', {
      path: '.',
      assetHashType: AssetHashType.OUTPUT,
      bundling: {
        local: {
          tryBundle(outputDir: string, _: BundlingOptions): boolean {
            fs.writeFileSync(join(outputDir, 'hook.sh'), 'echo hello world');
            return true;
          },
        },
        // no fallback
        image: DockerImage.fromRegistry('public.ecr.aws/docker/library/busybox:stable'),
        command: ['exit 1'],
      },
    })
```
And then you would have a class like [Code](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Code.html) that gives you all the options.

Reply:
I'd need more help I'm afraid, inline code + base code class is one thing, but how exactly will it work with getting the value

Reply:
I'm not exactly sure what you mean by "value" here. Are you talking about getting the command to run into `Dockerfile`? We can maybe do something similar to the way [certificates are added](https://github.com/CloudSnorkel/cdk-github-runners/blob/206bed8c19b7c555f231d6306cbe0d15c08d4d63/src/providers/image-builders/codebuild.ts#L275-L280). It writes the file to a special folder and then `Dockerfile` looks for files in that folder and [only installs the certificates](https://github.com/CloudSnorkel/cdk-github-runners/blob/206bed8c19b7c555f231d6306cbe0d15c08d4d63/src/providers/docker-images/lambda/linux-arm64/Dockerfile#L10) if it's there. We can also try something else if you have a simpler idea.

If you want, we can do a quick video chat next week to discuss the details and figure out a good solution together. I'm also on the [CDK Slack](https://cdk.dev/) if you prefer chat.